internal/rpc/httpproxy.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2024 Canonical.
+
+package rpc
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/juju/zaputil"
+	"github.com/juju/zaputil/zapctx"
+
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+)
+
+type HTTPClientOptions struct {
+	TLSConfig *tls.Config
```

Comment:
we could pass in an HTTP client instead?

Reply:
tls config must be used when creatinig the struct. I don't like the idea of creating the httpclient outside of the doRequest function, however i don't really have a strong opinion on that.
Why do you think it's better?

Reply:
Generally speaking you'd have clients specific for different purposes with different roundtrippers